tt_eager/tt_dnn/op_library/eltwise_unary/eltwise_unary_op.cpp

```diff
@@ -140,6 +143,16 @@ std::pair<string, string> get_op_init_and_func_parameterized(
         case UnaryOpType::UNARY_NE: op_init_and_name = {"unary_ne_tile_init();", fmt::format("unary_ne_tile({}, {}u);", idst, Converter::to_hex(param0))}; break;
         case UnaryOpType::UNARY_GT: op_init_and_name = {"unary_gt_tile_init();", fmt::format("unary_gt_tile({}, {}u);", idst, Converter::to_hex(param0))}; break;
         case UnaryOpType::UNARY_LT: op_init_and_name = {"unary_lt_tile_init();", fmt::format("unary_lt_tile({}, {}u);", idst, Converter::to_hex(param0))}; break;
+        case UnaryOpType::SOFTPLUS: {
+            TT_ASSERT(params.size() == 2, "Expected softplus to take 2 parameters");
+            float param1 = params[1];
+            op_init_and_name = {"softplus_tile_init();", fmt::format("softplus_tile({}, {}u, {}u, {}u);", idst,
```

Comment:
I see it is the way here, but I wonder, why do we bake param values into a string?

Reply:
Not sure since I'm also just looking at this for the first time, but I think it is just a cheap and dirty way to get them in the binary at compile time.